bindings/rust/src/joypad_nintendo.rs

```diff
@@ -0,0 +1,79 @@
+use std::ffi::{c_int, c_void};
+use crate::c_bindings::{
+    inputtino_joypad_switch_create,
+    inputtino_joypad_switch_destroy,
+    inputtino_joypad_switch_get_nodes,
+    inputtino_joypad_switch_set_on_rumble,
+    inputtino_joypad_switch_set_pressed_buttons,
+    inputtino_joypad_switch_set_stick,
+    inputtino_joypad_switch_set_triggers,
+};
+use crate::common::{DeviceDefinition, error_handler_fn};
+use crate::{get_nodes, make_device, JoypadStickPosition};
+
+pub struct SwitchJoypad {
+    joypad: *mut crate::c_bindings::InputtinoSwitchJoypad,
+    on_rumble_fn: Box<dyn FnMut(i32, i32) -> ()>,
+}
+
+impl SwitchJoypad {
+    pub fn new(device: &DeviceDefinition) -> Result<Self, String> {
```

Comment:
At a minimum, the error type should be `pub struct InputtinoError(String)`. Better would be an enum using `thiserror` or similar.